yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -903,6 +908,30 @@ object Client extends Logging {
   }
 
   /**
+   * Obtains token for the Hive metastore and adds them to the credentials.
+   */
+  private def obtainTokenForHiveMetastore(conf: Configuration, credentials: Credentials) {
+    if (UserGroupInformation.isSecurityEnabled /* And Hive is enabled */) {
```

Comment:
So, I think the `/* And Hive is enabled */` part is kinda important, and not talking just about Hive being compiled in. What would happen if a user doesn't have any Hive services running?

I'm also a little worried about precedent here... couldn't we make the same argument for acquiring HBase tokens? Solr? Somthing else? Unfortunately, I can't think of any alternative off the bat, given the way cluster mode works... :-/